### ISSUE TITLE
Uploading of large files/archives through Python API results in ConnectionAbortedError [WinError 10054]

### ISSUE THREAD

**Issue by PeterBetlem** [zenodo/zenodo#2328]
## Setup
Python API; following https://developers.zenodo.org/#quickstart-upload with new files API

## Describe the bug
Uploads using request.put() are randomly and abruptly stopped anywhere between being 10 and 99% uploaded.
Not as frequently an issue for smaller files (as they appear to finish the uploading process enough), but larger files (e.g., 25 GB zip file) are often affected as connections are reset even when 99% of the data has been uploaded.

Similar-sized (large) files also tend to result in upload errors when using the zenodo.org frontend/GUI, leaving no fool-proof way of uploading these files...

## Steps to Reproduce
Follow Python API documentation at https://developers.zenodo.org/#quickstart-upload with new files API.

## Expected behavior
- [ ] Uploading works and does not lead to ConnectionResetErrors...
- [ ] A possibility to resume the uploading process after the above error is thrown? Would save a lot of time and bandwidth.

## Additional context
Tried and failed:
- requests.put()
```
with open(filepath, "rb") as fp:
    r = requests.put(f"{url}/{filepath.name}", data = fp, params = {"access_token":xxx})
```
- requests sessions, session.put()
```
s = requests.Session()
with open(filepath, "rb") as fp:
     r = s.put(f"{url}/{filepath.name}", data = fp, params = {"access_token":xxx})
```
- requests.put() with chunk encoding (as per https://2.python-requests.org/en/master/user/advanced/#chunk-encoding)
```
def read_in_chunks(file_object, blocksize=1024):
    """Lazy function (generator) to read a file piece by piece.
    Default chunk size: 1k."""
    
    while True:
        data = file_object.read(blocksize)
        if not data:
            break
        yield data

with open(filepath, "rb") as fp:
    r = s.put(f"{url}/{filepath.name}", data = read_in_chunks(fp, filepath.stat().st_size), params = {"access_token":xxx})
```

Interestingly - Chunk Encoding sometimes  works to reduce the ConnectionReset frequency for larger files, which then manage to more frequently fully complete. However, small files (e.g., tested up to 200 MB) result in a HTTP Error 400, 'message': 'File is smaller than expected.' error. What could be the cause of this?

**Comment by asross** [comment]
I'm having similar issues -- I couldn't upload a 10GB file via the web interface, and when I tried it with the API, I either got ConnectionReset or "File is smaller than expected." Right now it just doesn't seem possible to upload large files to Zenodo :(

**Comment by cailinw** [comment]
I had the same issues with uploading both a ~3GB file and a ~9GB file via the web interface. I was unable to use the new API, because the "bucket" key did not exist in the response from the API (when following instructions [here](https://developers.zenodo.org/#quickstart-upload)).

I found a workaround by using [zenodo-upload](https://github.com/jhpoelen/zenodo-upload) to upload my 9GB file, and I was able to upload in under 30 minutes after hours of struggling to use the web interface and Zenodo api. Maybe this might help others?

**Comment by ppanero** [comment]
@cailinw Could you contat us through our support channel, and send me the deposit id to which you tried to upload and an approximate times of when it happened? So I can further investigate?

**Comment by cailinw** [comment]
@ppanero Yes, I just contacted through the support contact form with details.

**Comment by DustinSokolowski** [comment]
Hey! 

Did this ever end up coming to a conclusion? I have been able to get up to 1Gb files uploaded after 10-15 attempts. I find that the time of day can sometimes determine if it stops or not (perhaps the servers are too busy?) However, I have 5-6 files that are giving me considerable trouble. A more consistent solution would be much appreciated.

Best!
Dustin

**Comment by psybers** [comment]
Is there no update on this?  It seems like it is pretty much impossible to upload larger files to Zenodo.

**Comment by ppanero** [comment]
@psybers while the situation has improved a bit, we are still working on the fixes.

**Comment by harisushehu** [comment]
Hi @ppanero I am trying to upload a file that is about 80 GB in size and I am having the same problem using both web interface and [zenobo-upload](https://github.com/jhpoelen/zenodo-upload). Is there any progress regarding this?

**Comment by ppanero** [comment]
@harisushehu I'm not working at Zenodo anymore. I'd suggest opening a support ticket.

**Comment by harisushehu** [comment]
@ppanero Thanks for your response. I figured that breaking the data into smaller chunks works for me.
